src/utils/__tests__/timezones.test.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * load locale data manually that is not the current locale so that
+ * we can use it for unit testing
+ */
+
+import germanLocaleData from '@box/cldr-data/locale-data/de-DE';
+import russianLocaleData from '@box/cldr-data/locale-data/ru-RU';
+import japaneseLocaleData from '@box/cldr-data/locale-data/ja-JP';
+import timezones, { findTimeZone, TimeZone, TimeZoneEntries } from '../timezones';
+
+const germanZoneData = germanLocaleData.timezones;
+const russianZoneData = russianLocaleData.timezones;
+const japaneseZoneData = japaneseLocaleData.timezones;
+
+describe('util/timezones', () => {
+    test('should work in English', () => {
+        const zones: TimeZoneEntries = timezones({
+            includeZones: 'all',
+        }); // default is English
+        expect(zones['1']).toStrictEqual({
+            name: 'Africa/Abidjan',
+            displayName: 'GMT+00:00 Africa/Abidjan GMT',
+            id: 1,
+        });
+        expect(zones['139']).toStrictEqual({
+            name: 'America/Los_Angeles',
+            displayName: 'GMT-08:00 America/Los_Angeles PST',
```

Comment:
What about when we are in daylight saving time? Would it return `GMT-07:00 America/Los Angeles PDT`?

Reply:
As far as I can tell, this list come from the php `DateTimeZone` class in php from the monolith source file `./modular/i18n/i18n/legacy/lib/Box/Timezone.php`. I think what it is doing is checking today's date to find the current offset for each of these time zones as it constructs the list dynamically. We don't have that functionality available in React/JS unless we somehow successfully integrate ilib which has it.

Some possible solutions:

1. Do the work of finishing the ilib integration, including fixing the ilib webpack plugins to get all of the time zone json files into the webpack. This has footprint implications, and is a heavy-weight solution to a small problem. Plus it may take a sprint or two more, which admin-console cannot wait for.
2. Extract the parts of ilib that do the timezone work into a separate project, and use it box-ui-elements (and ilib too). That way we can avoid the ilib webpack plugin problem. These parts are relatively independent of the rest of ilib, so it is possible, especially if we abandon the idea of multi-calendar-type support. (ie. only support Gregorian). Also, put the tz data into that module as well so that this new code has the knowledge of when and if DST starts and ends for each zone, and we can make a query "what is the offset for this time zone at this datetime". This is also a heavy-weight solution, though ultimately more useful than #1.
3. Punt on the dynamic idea and just make strings like:

`GMT-08:00 PST/-07:00 PDT America/Los_Angeles`

That is, for those zones that use DST, put a joint string. For those string that do not, just put something like:

`GMT-07:00 MST America/Phoenix`

This is easy and quick to do, but is slightly less useful to the user.

Reply:
Or we can #3 now, and #2 later?

Reply:
Or, we can add:
4. Extract the parts of ilib that do the timezone work and put them directly into BUIE. This presents no integration problems and we could jettison the non-Gregorian parts of it which simplifies it a bunch, but has the problems that it increases the webpack footprint and would have to be maintained separately from ilib.

Reply:
5. Rely on Intl.DateTimeFormat() which can do time zone calculations. Pros: fast, no code from ilib needs to be added to BUIE. Cons: does not work on IE 11. Alternate idea: use strategy #3 above for IE11, and Intl.DateTimeFormat if it is available.

Reply:
Decided to do strategy number 5 above with the fall back to static strings for IE or any other platform that doesn't support the latest Intl stuff.